dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFileOrchestrator.kt

```diff
@@ -52,6 +54,37 @@ internal class BatchFileOrchestrator(
     private var lastFileAccessTimestamp: Long = 0L
     private var lastCleanupTimestamp: Long = 0L
 
+    private val knownFiles: MutableSet<File> =
+        rootDir.listFilesSafe(internalLogger)?.toMutableSet() ?: mutableSetOf()
```

Comment:
I realized the it may bring the following concern: this will be called on the user thread during SDK initialization and since it is I/O call, it may slow down SDK start (especially if there is a lot of files on the disk).

Instead, we maybe can to initial read in the `getWritableFile` / `getReadableFile` methods (well, on any initial access of this property), this will shift it to the worker thread improving SDK initialization.

Same may apply to the `root` folder creation.